server/webapp/WEB-INF/rails.new/app/assets/javascripts/pipeline_errors.js

```diff
@@ -0,0 +1,62 @@
+/*************************GO-LICENSE-START*********************************
+ * Copyright 2014 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *************************GO-LICENSE-END**********************************/
+
+PipelineErrors = function ($) {
+    var refreshPipelineErrors = function() {
+        var pipelinesHaveError = [];
+        $("#cruise_message_body [data-pipelines]").each(function (i, ele) {
+            if ($(ele).data("pipelines")) {
+                var pipelines = $(ele).data("pipelines").split(",");
+                pipelinesHaveError = pipelinesHaveError + pipelines;
+            }
+        });
+        $(".pipeline").each(function (i, pipeline) {
```

Comment:
@xli have to tried doing this on a config with more than 500 pipelines? This is going to run very 5 seconds according to  ```DashboardPeriodicalExecuter``` I suspect this will block the thread for too long.